### PR TITLE
TDD determine check, obstructed? fixes, and adding default value to captured

### PR DESCRIPTION
**Pull request by icrustandi**
### What does this PR do?
1. Check? method  returns a boolean that indicates whether or not the current state of the game is in check during a player's turn. 
2. Added tests for obstructed? method in piece.rb and subsequently found mistakes in the method.
3. Add default value false to captured column in pieces table.

### Where should I start?
**app/models/game.rb**
-   Modified populate board to include 'captured: false' attribute when creating starting pieces
-   Added check? method

**app/models/knight.rb**
-  Modified the name of legal_move? method to valid_move? method to be consistent with the other pieces

**spec/models/knight_spec.rb**
-   Modified knight test to reflect the name change above

**spec/factories/games.rb**
-   Added a factory to create a board with a pieces in a check state

-   Added a factory to test board with obstacles

**spec/models/game_spec.rb**
-   Added RSpec test to test check method. Might need to add more robust tests for this

**spec/models/piece_spec.rb**
-   Added tests to Rspec for obstructed?

**app/models/piece.rb**

-   Modified x_diff and y_diff to absolute value distances

-   Removed [x1,y1] in places_between since destination coordinates are not included in the obstructed? scope

-   Included a case where an input might not be a horizontal/vertical/diagonal move, which should raise an error

**db/migrate/20161114184117_add_default_to_captured.rb**

-   Migration file to add default value false to captured column

**db/schema.rb**
-   Changes in migration file are reflected here

### How do I manually test this?
While in the pancakes project folder, run the following in console:

1. rspec 'spec/models/game_spec.rb'
2. rspec 'spec/models/piece_spec.rb'
3. You may need to run rake db:migrate and rake db:migrate RAILS_ENV=test

### CODE REVIEW

File: spec/factories/games.rb [hunk 1]
```diff
@@ -13,5 +13,50 @@
       white_player_id 1
       black_player_id 2
     end
+
+    trait :check_scenario do
```
Comment (danielpowell4) [inline spec/factories/games.rb]: We should likely have tests that check a knight-puts-king-into check scenario as it's different.

Good to check x-attack, y-attack and diagonal attacks as well

**Comment by danielpowell4** [comment]
Left a small commit just to bolster the `sure it works`-ness of the check method.

Should be ready-to-roll soon though!

File: spec/factories/games.rb (hunk 1 above)
Comment (icrustandi) [inline spec/factories/games.rb]: @danielpowell4 Thanks! Good suggestions - I've added tests for vertical, diagonal, and knight checks

**Comment by icrustandi** [comment]
@danielpowell4 Added the commit for the new tests! Should I merge?

**Comment by danielpowell4** [comment]
Looks great.

Cool this is starting to come together.

Merge when you're ready!

**Pull request merged**